front/app/modules/commercial/content_builder/admin/components/CraftComponents/Iframe/index.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+
+// components
+import { Box, IconTooltip, Input } from '@citizenlab/cl2-component-library';
+import Error from 'components/UI/Error';
+
+// intl
+import messages from '../../../messages';
+import { injectIntl } from 'utils/cl-intl';
+
+// craft
+import { useNode } from '@craftjs/core';
+
+// events
+import eventEmitter from 'utils/eventEmitter';
+
+interface Props {
+  url: string;
+  height: number;
+  hasError: boolean;
+  errorType?: string;
+  title?: string;
+}
+
+const Iframe = ({ url, height, hasError, title }: Props) => {
+  return (
+    <Box minHeight="26px">
+      {!hasError && url && (
+        <iframe src={url} title={title} width="100%" height={height} />
+      )}
+    </Box>
+  );
+};
+
+const IframeSettings = injectIntl(({ intl: { formatMessage } }) => {
+  const {
+    actions: { setProp },
+    url,
+    height,
+    id,
+    hasError,
+    errorType,
+    title,
+  } = useNode((node) => ({
+    url: node.data.props.url,
+    height: node.data.props.height,
+    id: node.id,
+    title: node.data.props.title,
+    hasError: node.data.props.hasError,
+    errorType: node.data.props.errorType,
+  }));
+
+  return (
+    <Box flexWrap="wrap" display="flex" gap="16px" marginBottom="20px">
+      <Box flex="0 0 100%">
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeUrlLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeUrlLabelTooltip)}
+              />
+            </span>
+          }
+          placeholder={formatMessage(messages.iframeUrlPlaceholder)}
+          type="text"
+          value={url}
+          onChange={(value) => {
```

Comment:
I would change this from an inline function to a function stored in a variable, e.g. 

```
const handleChange = (value) => ...

onChange={handleChange}
```

Also think the `setProps((props) => (props.url = value))` stuff is a bit strange. Good practice in React (and in general) is to keep functions as pure as possible. Mutating a property of an object that you receive as a function argument is a kind of side effect you really want to avoid. Better would be something like

```
setProp((props) => ({ ...props, url: value }))
```

Also I don't know how craft.js works exactly but maybe this could all be combined in one statement?

```
setProps((props) => ({ ...props, url: value, errorType: validation[1], hasError: !validation[0] }))
```

Reply:
Thanks Luuc - your suggestions are definitely more concise. I lifted out the code into a handleChange function like you recommended, however it seems that this strange format `setProp((props) => (props.url = value))` is required by craft. I tried a few different variations, but this seems like the only way. I looked into the craft.js library code as well and that's what's being used there. I'll make a note of this and check with Iva when she is back though too :)